packages/ozone/src/mod-service/views.ts

```diff
@@ -309,39 +318,46 @@ export class ModerationViews {
   }
 
   async recordDetail(
-    subject: RecordSubject,
+    subjects: RecordSubject[],
     labelers?: ParsedLabelers,
-  ): Promise<RecordViewDetail | undefined> {
-    const [records, subjectStatusesResult] = await Promise.all([
-      this.records([subject]),
-      this.getSubjectStatus([subject.uri]),
-    ])
-    const record = records.get(subject.uri)
-    if (!record) return undefined
+  ): Promise<Map<string, RecordViewDetail>> {
+    const subjectUris = subjects.map((s) => s.uri)
+    const [records, subjectStatusesResult, localLabels, externalLabels] =
+      await Promise.all([
+        this.records(subjects),
+        this.getSubjectStatus(subjectUris),
+        this.labels(subjectUris),
+        this.getExternalLabels(subjectUris, labelers),
+      ])
+
+    const results = new Map<string, RecordViewDetail>()
+
+    records.forEach(async (record, uri) => {
```

Comment:
Pretty sure you lose track of the promise like this. I think you either need to use a for loop or `map()` with a `Promise.all`. 

I'd be inclined to do the latter so you can grab the blob infos in parallel